common/common.go

```diff
@@ -93,3 +94,67 @@ func NanosToMillis(nanos int64) int64 {
 func TimeToMillis(ts time.Time) int64 {
 	return NanosToMillis(ts.UnixNano())
 }
+
+// OffsetsBySource is a map of wal Offsets keyed to source ids
+type OffsetsBySource map[int]wal.Offset
```

Comment:
This new type appears in a lot of places. Basically, in places where we used to track a single offset, we now track a map of offsets keyed to source IDs.